python/ray/air/tests/test_resource_changing.py

```diff
@@ -0,0 +1,140 @@
+from ray.air import session
+from ray.air.checkpoint import Checkpoint
+from ray.air.config import FailureConfig, RunConfig
+from ray.air.constants import TRAIN_DATASET_KEY
+from ray.tune.tune_config import TuneConfig
+from ray.tune.tuner import Tuner
+from ray.train.data_parallel_trainer import DataParallelTrainer
+from ray.train.xgboost import XGBoostTrainer
+from sklearn.datasets import load_breast_cancer
+import pandas as pd
+import pytest
+import ray
+from ray import tune
+from ray.tune.schedulers.resource_changing_scheduler import (
+    DistributeResources,
+    ResourceChangingScheduler,
+)
+from ray.tune.schedulers.async_hyperband import ASHAScheduler
+
+
+@pytest.fixture
+def ray_start_8_cpus():
+    address_info = ray.init(num_cpus=8)
+    yield address_info
+    # The code after the yield will run as teardown code.
+    ray.shutdown()
+
+
+def train_fn(config):
+    start_epoch = 0
+
+    print(session.get_trial_resources())
+    checkpoint = session.get_checkpoint()
+    if checkpoint:
+        # assume that we have run the session.report() example
+        # and successfully save some model weights
+        checkpoint_dict = checkpoint.to_dict()
+        start_epoch = checkpoint_dict.get("epoch", -1) + 1
+
+    # wrap the model in DDP
+    for epoch in range(start_epoch, config["num_epochs"]):
+        checkpoint = Checkpoint.from_dict(dict(epoch=epoch))
+        session.report(
+            {"metric": config["metric"] * epoch, "epoch": epoch}, checkpoint=checkpoint
+        )
```

Comment:
I know we have tests for the resource changing scheduler in Tune. However, for the sanity of these tests, can we report `trial_resources.get("CPU")` here and later check that they are different for at least one trial?